influxdbconnector/src/main/java/org/openbase/bco/app/influxdbconnector/InfluxDbconnectorApp.java

```
@@ -22,48 +22,255 @@
+    private static final Integer READ_TIMEOUT = 60;
+    private static final Integer WRITE_TIMEOUT = 60;
+    private static final Integer CONNECT_TIMOUT = 40;
+
 
     public InfluxDbconnectorApp() throws InstantiationException {
 
+
+
     }
 
     @Override
-    protected ActionDescription execute(ActivationState activationState) throws CouldNotPerformException, InterruptedException {
-        logger.info("execute influx db connector");
+    protected ActionDescription execute(ActivationState activationState) throws CouldNotPerformException {
+        try {
+            databaseUrl = generateVariablePool().getValue("INFLUXDB_URL");
+            databaseName = generateVariablePool().getValue("INFLUXDB_NAME");
+            batchTime = Integer.valueOf(generateVariablePool().getValue("INFLUXDB_BATCH_TIME"));
+            batchLimit = Integer.valueOf(generateVariablePool().getValue("INFLUXDB_BATCH_LIMIT"));
+        } catch (NotAvailableException ex) {
+            ExceptionPrinter.printHistory(ex, logger);
+        }
+
+
+
         task = GlobalCachedExecutorService.submit(() -> {
-            while (!task.isCancelled()) {
-                try {
-                    logger.info("connect to db at: "+ generateVariablePool().getValue("INFLUXDB_URL" ));
-                    Thread.sleep(5000);
-                } catch (InterruptedException | CouldNotPerformException ex) {
-                    ExceptionPrinter.printHistory(ex, logger);
+            logger.info("execute influx db connector");
+            try {
+                if (initiateDatabase()) {
+
+                    customUnitPool = new CustomUnitPool();
+
+                    unitStateObserver = (source, data) -> saveInDB((Unit) source.getServiceProvider(), source.getServiceType(), data);
+
+
+                    init();
                 }
+
+            } catch (InstantiationException | InitializationException | InterruptedException ex) {
```

Please do not catch Interrupted Exceptions without forwarding it or recovering the interrupted state.
`Thread.currentThread().interrupt();  // set interrupt flag`

---

At least we want to let the task fail if interrupted.